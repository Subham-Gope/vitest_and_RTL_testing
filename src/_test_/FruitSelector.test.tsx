// import { describe, test, expect } from "vitest";
// import { screen, render } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import FruitSelector from "../components/FruitSelector";

// describe("Testing the FruitSelector.tsx file", () => {
//   test("selects and deslects options", async () => {
//     // setting up the userEvent
//     const user = userEvent.setup();

//     // setting up the debugger
//     screen.debug();

//     // setting up the snapshot matching
//     const { container } = render(<FruitSelector />);
//     expect(container).toMatchSnapshot();

//     const select = screen.getByRole("combobox", {
//       name: /select the fruit/i,
//     }) as HTMLSelectElement;
//     expect(select).toBeInTheDocument();

//     // Initiating user interactions
//     await user.selectOptions(select, "bannana");
//     expect(select).toHaveValue("bannana");

//     await user.selectOptions(select, "Apple");
//     expect(select).toHaveValue("apple");

//     await user.selectOptions(select, "pear");
//     expect(select).toHaveValue("pear");

//     await user.selectOptions(select, "jackfruit");
//     expect(select).toHaveValue("jackfruit");
//   });
// });
