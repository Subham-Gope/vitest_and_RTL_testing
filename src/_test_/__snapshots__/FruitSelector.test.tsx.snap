// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the FruitSelector.tsx file > selects and deslects options 1`] = `
<div>
  <div>
    <label
      for="fruits"
    >
      Select the Fruit you want:
    </label>
    <select
      id="fruits"
    >
      <option
        disabled=""
        value=""
      >
        Select
      </option>
      <option
        value="bannana"
      >
        Bannana
      </option>
      <option
        value="apple"
      >
        Apple
      </option>
      <option
        value="pear"
      >
        Pear
      </option>
      <option
        value="jackfruit"
      >
        Jack Fruit
      </option>
    </select>
  </div>
</div>
`;
